lib/assets/javascripts/utils/filteriseTable.js

```diff
@@ -0,0 +1,73 @@
+/* 
```

Comment:
There is no need of a map of debounced functions, regex.match is computationally more expensive than test, there is not need for checking query.length. See my proposal below

/* 
  filteriseTable adds filter capabilities to an HTML table 
  table rows are shown/hidden as the user enters text into the filter input field 
  all rows are made visible when the user clicks the 'clear' icon
*/
(function(ctx){

  var filter = (function(el){
    var query = $(el).val();
    var regex = new RegExp(query, 'i');
    $.each($(el).closest("table").find("tbody tr"), function(idx, tr){
        if(regex.test($(tr).text())){
          $(tr).show();
        }else{
          $(tr).hide();
        }
      });
  });

  var clear = (function(el){
    $(el).val('');
    $(el).closest("table").find("tbody tr").show();
  });

  ctx.init = ctx.init || (function(options){
    if($ && options && options.selector){
      var debounced = dmproadmap.utils.debounce(filter);
      $(options.selector).keyup(function(e){
        debounced(options.selector);
      });

      $(options.selector).siblings("#clear_filter").click(function(e){
        e.preventDefault();
        clear(options.selector);
        debounced.cancel();
      });
    }
  });

})(define('dmproadmap.utils.filteriseTable'));